src/Database/LSMTree/Internal.hs

```diff
@@ -1634,4 +1634,9 @@ supplyUnionCredits t credits = do
       RW.withWriteAccess (tableContent tEnv) $ \tableContent ->
         case tableUnionLevel tableContent of
           NoUnion -> pure (tableContent, credits) -- all leftovers
-          Union{} -> error "supplyUnionCredits: not yet implemented"
+          Union{}
+           | credits <= UnionCredits 0 -> pure (tableContent, UnionCredits 0)
+           --TODO: remove this 0 special case once the general case covers it.
+           -- We do not need to optimise the 0 case. It is just here to
+           -- simplify test coverage.
+           | otherwise -> error "supplyUnionCredits: not yet implemented"
```

Comment:
I figured we would more reliably notice this TODO and resolve it when we implement the general case, rather than notice a TODO on an out-of-the-way unit test to say to enable it once `supplyUnionCredits` was implemented. So better to have the unit test running, and temporarily implement the special case in the real impl.